### PR TITLE
Remove code that is no longer needed when using PHP >= 5.5.0

### PR DESCRIPTION
**Pull request by atrol**
Fixes #23381

### CODE REVIEW

**Comment by vboctor** [comment]
@atrol If an instance has database type as `mysql` do we still support that and map it to `mysqli`?

**Comment by atrol** [comment]
> If an instance has database type as mysql do we still support that and map it to mysqli?

No, it's no longer supported, there is no mapping.
I added a check in install.php to tell users to use mysqli instead of mysql.

File: admin/install.php [hunk 1]
```diff
@@ -246,9 +246,15 @@ function print_test( $p_test_description, $p_result, $p_hard_fail = true, $p_mes
...
 			print_test( 'Checking PHP support for Microsoft SQL Server driver',
-				version_compare( phpversion(), '5.3' ) < 0, true,
+				BAD, true,
 				'mssql driver is no longer supported in PHP >= 5.3, please use mssqlnative instead' );
 		}
+
+		if( $f_db_type == 'mysql' ) {
+			print_test( 'Checking PHP support for MySQL driver',
+				BAD, true,
+				'mysql driver is deprecated in PHP >= 5.5, please use mysqli instead' );
```
Comment (dregad) [inline admin/install.php]: I would add that we no longer support it.

Comment (atrol) [reply admin/install.php]: Changed the message.

**Pull request merged**